pkg/database/sqlserver.go

```diff
@@ -0,0 +1,208 @@
+package database
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/fraenky8/tables-to-go/pkg/settings"
+
+	// sqlserver database driver
+	_ "github.com/denisenkom/go-mssqldb"
+)
+
+// Sqlserver implements the Database interface with help of GeneralDatabase.
+type Sqlserver struct {
+	*GeneralDatabase
+
+	defaultUserName string
+}
+
+// NewSqlserver creates a new Sqlserver database.
+func NewSqlserver(s *settings.Settings) *Sqlserver {
+	return &Sqlserver{
+		GeneralDatabase: &GeneralDatabase{
+			Settings: s,
+			driver:   dbTypeToDriverMap[s.DbType],
+		},
+		defaultUserName: "sa",
+	}
+}
+
+// Connect connects to the database by the given data source name (dsn) of the
+// concrete database.
+func (ss *Sqlserver) Connect() error {
+	return ss.GeneralDatabase.Connect(ss.DSN())
+}
+
+// DSN creates the DSN String to connect to this database.
+func (ss *Sqlserver) DSN() string {
+	user := ss.defaultUserName
+	if ss.Settings.User != "" {
+		user = ss.Settings.User
+	}
+	return fmt.Sprintf("server=%s;port=%s;user=%s;database=%s;password=%s",
+		ss.Settings.Host, ss.Settings.Port, user, ss.Settings.DbName, ss.Settings.Pswd)
+}
+
+// GetTables gets all tables for a given schema by name.
+func (ss *Sqlserver) GetTables() (tables []*Table, err error) {
+
+	err = ss.Select(&tables, `
+		SELECT
+            table_name
+        FROM
+            INFORMATION_SCHEMA.TABLES
+        WHERE
+            TABLE_SCHEMA = @p1
+        ORDER BY TABLE_NAME
+	`, ss.Schema)
+
+	if ss.Verbose {
+		if err != nil {
+			fmt.Println("> Error at GetTables()")
+			fmt.Printf("> schema: %q\r\n", ss.Schema)
+		}
+	}
+
+	return tables, err
+}
+
+// PrepareGetColumnsOfTableStmt prepares the statement for retrieving the
+// columns of a specific table for a given database.
+func (ss *Sqlserver) PrepareGetColumnsOfTableStmt() (err error) {
+
+	ss.GetColumnsOfTableStmt, err = ss.Preparex(`
```

Comment:
Just to be double sure, since I don't have any experience with SQLServer - this statement is the same as for PG except the placeholder format?